### PR TITLE
User story 21 - User Can Edit their Password

### PR DESCRIPTION
**Pull request by tylertomlinson**
## All Submissions:

* [x] Merge the target branch into the PR branch. Fix any conflicts that might appear.
* [x] Have you checked to ensure that other pull request are approved for merge [Pull Requests](../../../pulls)?
* [x] What stories does the PR include

### New Feature Submissions:

* [x] Have you added an explanation of what your changes do?
* [x] Have you lint your code locally prior to submission?

### Testing:

* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?
* [x] All new and existing tests passed.

**Test coverage** = 100%
--------------------------------------------------------------------------------
### Change Log

Created test for editing user password with edge cases (blank fields)
Created `edit` and `update` password on user controller 
Added routes for above

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -35,6 +31,22 @@ def update
...
   end
 
+  def update_password
+    user = current_user
+    user.update(user_params)
+    if user.save
+      flash[:notice] = "Your password has been updated."
+      redirect_to '/profile'
+    else
+      flash[:error] = user.errors.full_messages.to_sentence
```
Comment (alex-latham) [inline app/controllers/users_controller.rb:41]: The errors method has been so useful!

**Review by alex-latham — APPROVED** [review]
I like these changes, and don't see any issues preventing approval.

**Review by danielpselph — APPROVED** [review]
Tests are clean and easy to follow, changes approved.
